tools/unirestore/scylla_restore.py

```diff
@@ -0,0 +1,996 @@
+    schema = parse_schema_file(config['schema_file'])
+
+    # Get list of cluster node IPs
+    nodes = get_cluster_hosts(config['dest_host'], ssh_auth)
+    logger.info('Cluster node list: %s', str(nodes))
+
+    # Verify config "limit_to_nodes" setting
+    if 'limit_to_nodes' in config.keys():
+        if set(config['limit_to_nodes']).issubset(set(nodes)):
+            logger.info('Detected a limited subset of nodes to use: %s', str(config['limit_to_nodes']))
+            nodes = config['limit_to_nodes']
+            logger.info('will be using limited node set: %s', str(nodes))
+        else:
+            logger.error('Nodes specified in configuration are not present in destination cluster')
+            sys.exit(1)
+
+    # Ensure schema versions are up to date
+    logger.info('Verifying schema version in sync')
+    check_schema_version(nodes, ssh_auth)
+
+    # Find out the destination directories from the cluster
+    dest_dirs = get_destination_dirs(schema, config, ssh_auth)
+
+    # Check the ssh user has write permissions in the directories
+    logger.info('Verifying node directories are writeable:')
+    for dest_dir in dest_dirs:
+        success = verify_scp_possible(nodes, dest_dir, ssh_auth=ssh_auth)
+        if success:
+            logger.info('- %s', dest_dir)
+        # no need for else because the check will exit on verify_scp_possible()
+
+    # Read the token ranges host affinity from the database per keyspace
+    #ring = {}
+    #for ks in schema.keys():
+    #    ring[ks] = get_ks_ranges(config['dest_host'], ks, ssh_auth)
+
+    # Map restored files
+    file_list = get_toc_file_list(config['backup_root_path'], schema)
+    src_nodes = get_src_nodes(file_list)
+    uidgid = config['scylla_user'] + ':' + config['scylla_group']
+
+    # check for resume or new restore plan
+    if resume:
+        rplan = resume
+    else:
+        # Build a new restore plan
+        #restore_plan = build_restore_plan(file_list, ring, dest_dirs, src_nodes, nodes)
+        restore_plan = build_restore_plan(file_list, dest_dirs, src_nodes, nodes)
+        rplan = save_restore_plan(timestamp, restore_plan)
+
+    # Run restore
+    execute_restore_plan(nodes,schema, rplan, config, uidgid, file_list, ssh_auth)
+
+    logger.info('Finished copying restore files')
+    logger.info('Restore complete')
+
+# root
+args = get_args()
+
+# logging configuration - double handlers for both file and stdout logging.
+logging.basicConfig(format="%(asctime)s::%(name)s::%(levelname)s::%(message)s",
+                            level=args.loglevel,
+                            handlers=[logging.FileHandler("scylla_restore.log"),
+                                      logging.StreamHandler(sys.stdout)])
+logger = logging.getLogger(__name__)
+logger.info('Starting restore')
+
+if args.resume and os.path.isfile(args.resume):
+    logger.info("Retrying from %s", args.resume)
+    main(args.config, args.resume)
+else:
+    main(args.config, None)
```

Comment:
better to put the main code under such if, if ever want to import this file, for example in testing
```suggestion
if __name__ == "__main__":
    logger.info('Starting restore')

    if args.resume and os.path.isfile(args.resume):
        logger.info("Retrying from %s", args.resume)
        main(args.config, args.resume)
    else:
        main(args.config, None)
```

Reply:
I meant to do this, thanks for the reminder (I was sure I put a TODO to do it)